packages/cli/src/services/hooks.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Service } from 'typedi';
+import { UserService } from '@/services/user.service';
+import type { AssignableRole, User } from '@/databases/entities/User';
+
+/**
+ * Exposes functionality to be used by the cloud BE hooks.
+ * DO NOT DELETE or RENAME any of the methods without making sure this is not used in cloud BE hooks.
+ */
+@Service()
+export class HooksService {
+	constructor(private userService: UserService) {}
+
+	/**
+	 * Invites users to the instance.
+	 * This method is used in the cloud BE hooks, to invite members during sign-up
+	 */
+	async inviteUsers(owner: User, attributes: Array<{ email: string; role: AssignableRole }>) {
```

Comment:
There are plenty more function calls in the hooks that call into some code in the core app. Things like `issueCookie`, calls to `SettingsRepository`, and all of the owner initialization code in `setupOwner`.
We should add of those in here as well, otherwise I don't see much of a point in adding just one method here.